experimental/dds/tree2/src/test/simple-tree/utils.ts

```diff
@@ -36,6 +54,33 @@ export function getRoot<TRoot extends TreeFieldSchema>(
 	return view.root;
 }
 
+/**
+ * Helper for making small test schemas.
+ */
+export function makeSchema<TSchema extends ImplicitFieldSchema>(
+	fn: (factory: SchemaFactory<string>) => TSchema,
+) {
+	return fn(new SchemaFactory(`test.schema.${Math.random().toString(36).slice(2)}`));
+}
+
+/**
+ * Given the schema and initial tree data, returns a hydrated tree node.
+ */
+export function getRoot<TSchema extends ImplicitFieldSchema>(
+	schema: TSchema | ((factory: SchemaFactory<string>) => TSchema),
+	data: InsertableTreeFieldFromImplicitField<TSchema>,
+): TreeFieldFromImplicitField<TSchema> {
+	if (typeof schema === "function") {
+		// eslint-disable-next-line no-param-reassign
+		schema = makeSchema(schema as (builder: SchemaFactory<string>) => TSchema);
+	}
+	const config = new TreeConfiguration(schema, () => data);
+	const factory = new TreeFactory({});
```

Comment:
We can reuse the same factory since its stateless (put it outside this method, and maybe export it)

Also since this is our  internal test stuff, we might as well opt into more error checking by configuring our factor with a json validator and the reference forest.